### PR TITLE
[Lang] Limit non-first division of an axis on a SNodeTree path to a power of two

### PR DESCRIPTION
**Pull request by strongoier**
Issue: #6660

### CODE REVIEW

File: taichi/ir/snode.cpp [hunk 1]
```diff
@@ -54,30 +54,26 @@ SNode &SNode::create_node(std::vector<Axis> axes,
...
-    auto &ind = axes[i];
-    new_node.extractors[ind.value].activate(
+    int ind = axes[i].value;
+    auto end = new_node.physical_index_position + new_node.num_active_indices;
+    bool is_first_division =
+        std::find(new_node.physical_index_position, end, ind) == end;
+    if (is_first_division) {
+      new_node.physical_index_position[new_node.num_active_indices++] = ind;
+    } else {
+      if (!bit::is_power_of_two(sizes[i])) {
```
Comment (jim19930609) [inline taichi/ir/snode.cpp:64]: Shall we limit this check to packed mode only? In non-packed mode, we always pad `sizes` to the power of 2, so this limitation should've never applied?

**Comment by strongoier** [comment]
According to the discussions in #6660, I've turned the error into a warning.

> Interesting... If I understand it correctly, after this PR, the packed mode and non-packed mode will only influence the first axis of a snode tree?

More accurately, they will only influence the first division of an axis on a SNodeTree path.

File: taichi/ir/snode.cpp (hunk 1 above)
Comment (strongoier) [reply taichi/ir/snode.cpp]: You're right. Although non-packed mode will be removed, this PR should make a proper intermediate state.

**Review by jim19930609 — APPROVED** [review]
LGTM!

**Pull request merged**